batchconsumer/checkpointmanager.go

```diff
@@ -0,0 +1,88 @@
+package batchconsumer
+
+import (
+	"time"
+
+	kv "gopkg.in/Clever/kayvee-go.v6/logger"
+
+	"github.com/Clever/amazon-kinesis-client-go/batchconsumer/stats"
+	"github.com/Clever/amazon-kinesis-client-go/kcl"
+)
+
+type checkpointManager struct {
+	log kv.KayveeLogger
+
+	checkpointFreq time.Duration
+
+	checkpoint chan kcl.SequencePair
+	shutdown   chan chan<- struct{}
+}
+
+func NewCheckpointManager(
+	checkpointer kcl.Checkpointer, config Config, log kv.KayveeLogger,
+) *checkpointManager {
+	cm := &checkpointManager{
+		log: log,
+
+		checkpointFreq: config.CheckpointFreq,
```

Comment:
it would be somewhat easier to interpret this `New...` method if it took in just the checkpoint frequency instead of entire config object